test/bdd/pkg/didexchange/didexchange_controller_steps.go

```diff
@@ -357,20 +402,22 @@ func (a *ControllerSteps) performApprove(agentID string, useDID bool, connection
 		}
 	}
 
-	connections := struct{ Connections []string }{}
-
-	err := util.SendHTTP(http.MethodGet, controllerURL+"/mediator/connections", nil, &connections)
-	if err != nil {
-		return fmt.Errorf("mediator connections: %w", err)
-	}
-
 	logger.Debugf("Accepting invitation from controller for agent[%s], did[%s]",
 		agentID, publicDID)
 
+	if routerID == "" {
+		var err error
+
+		routerID, err = chooseConnection(controllerURL)
+		if err != nil {
+			return err
+		}
+	}
+
```

Comment:
Is router connection id mandatory to this controller API now ?

Reply:
No, it is not mandatory. But if you want to use `router` you need explicitly specify `router connection`.
The framework will not automatically choose the router. This is why we must provide the router ID.